libraries/Microsoft.Bot.Builder.Ai/LocaleConverterMiddleware.cs

```diff
@@ -48,22 +49,44 @@ public LocaleConverterMiddleware(Func<ITurnContext, string> getUserLocale, Func<
             {
                 if (!String.IsNullOrWhiteSpace(message.Text))
                 {
-                    string fromLocale = _getUserLocale(context);
-                    ConvertLocaleMessage(message, fromLocale);
-                    await _setUserLocale(context);
+                    bool localeChanged = await _setUserLocale(context);
+                    if (!localeChanged)
+                    {
+                        string fromLocale = _getUserLocale(context);
+                        ConvertLocaleMessage(context, fromLocale);
+                    }
+                    
                 }
             }
-            await next().ConfigureAwait(false);
+            if(!_isLastMiddleware)
```

Comment:
This approach isn't going to work, unfortunately. As things sit now, there is no concept of "isLast()". Changing this is a non-trivial DCR that would require signoff from a number of folks. 

The idea of "IsLast()" also doesn't really apply to the Russian Doll model that we're using. See the Express, Redux, or ASP.NEt Core Middleware models for more details of how this works.